src/plugins/security_oss/public/plugin.ts

```diff
@@ -51,8 +67,23 @@ export class SecurityOssPlugin
   }
 
   public start(core: CoreStart) {
+    const isAnonymousAccessEnabled = !!core.application.capabilities.security?.anonymousAccess;
```

Comment:
__note:__ My understanding and main assumption is that using endpoints and capabilities exposed by the X-Pack counterpart in the OSS version of the plugin is acceptable as long as it works fine with OSS distribution. What do you think?

Reply:
I think capabilities are more OK than endpoints, but I don't know if we've codified this anywhere.

I don't love the idea of using x-pack endpoints in OSS, because that could lead to `404` errors, which is sloppy. I see you've guarded against that here, but I worry about the precedent this sets.

For the insecure cluster warning, we defined the endpoint in OSS and allowed x-pack security to augment the logic. This allows the endpoint to work in all distributions, while still allowing for distribution-specific behavior.

Reply:
> I don't love the idea of using x-pack endpoints in OSS, because that could lead to 404 errors, which is sloppy. I see you've guarded against that here, but I worry about the precedent this sets.

> For the insecure cluster warning, we defined the endpoint in OSS and allowed x-pack security to augment the logic. This allows the endpoint to work in all distributions, while still allowing for distribution-specific behavior.

Yeah, from the technical side I agree with your sentiment. It just feels to me that we're building all these abstractions only to pretend that there is no explicit dependency between OSS and X-Pack plugin parts when in reality the only purpose of the the X-Pack_Something_OSS plugins is to support their main plugins across distributions while being useless on its own. The insecure cluster warning is just a great exception that does its job even without X-Pack part though.

Let me see if I can find a better way to preserve a status quo here :stuck_out_tongue: